### ISSUE TITLE
Allow saving "view" (set of tabs)

### ISSUE THREAD

**Issue by mattmazzola** [microsoft/AzureStorageExplorer#7468]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Problem

When I open Storage Explorer or Update version, it resets all of the tabs I have open. I work in various projects which have scripts that output to different known locations. I keep tabs open for each of these locations to easily verify operation

### Desired Solution

I would like to be able to save the current set of tabs I have open as a "view" with name. For example I can save a "Project A" set of tabs and each time I open explorer or update these tab will be restored

### Alternatives and Workarounds

Alternative is to manually reopen these tabs each time they are closed

**Comment by mattmazzola** [comment]
I would also suggest an option to control how much of the "path" is written to the tab name.

Currently only the "container" name is in the tab, but if you have multiple tabs directed to different folders within that container you cannot tell which is which.

![image](https://github.com/microsoft/AzureStorageExplorer/assets/2856501/6208676c-4d8e-4414-baa0-211995bd35d3)

In the figure below I have these four tabs open.
I would like option to get either "last folder name" or "full path"

| Container Name | Last Folder Name | Full Path |
| --- | --- | --- |
| (ambiguous) | (short, but low info) | (full info, but maybe too long for tab size?) |
| data | oa_figure | `/data/PMC_OpenAccessDataset/oa_figures/` |
| data | PMC_OpenAccessDataset | `/data/PMC_OpenAccessDataset/` |
| scratch | pmc_expansion | `/scratch/pmc_expansion/` |
| scratch | mattm | `/scratch/mattm/` |

**Comment by MRayermannMSFT** [comment]
> When I open Storage Explorer or Update version, it resets all of the tabs I have open. I work in various projects which have scripts that output to different known locations. I keep tabs open for each of these locations to easily verify operation

What OS are you on? Are you using a 3rd party package manager to manage your install? I would say that losing opened tabs on update is not expected.

**Comment by MRayermannMSFT** [comment]
> I would also suggest an option to control how much of the "path" is written to the tab name.
>
> Currently only the "container" name is in the tab, but if you have multiple tabs directed to different folders within that container you cannot tell which is which.

Yes, it has been on our minds for a while that this needs improvement. We should automatically be adding more "context" to the names of tabs as tabs with duplicate names appear. I'll open a dedicated feature request for this...

**Comment by mattmazzola** [comment]
> What OS are you on? Are you using a 3rd party package manager to manage your install?
```
Version: 1.31.2 (94)
AzCopy Version: 10.20.1

Platform: Windows
Architecture: x64 
Build Number: 20231003.6 
Commit: a71fade3e3 
Support ID: b4a338fd-5fc9-42d3-f511-c5d66e0b77cd 
```

> restore tab session enabled, yes?

I am not sure. I didn't know about this setting. If you meant the setting pictured below, I did not have it enabled.

![image](https://github.com/microsoft/AzureStorageExplorer/assets/2856501/cfdf4cbc-1456-43c8-a618-3c00579f088b)

I enabled it and this preserves the tabs on restart which is already major improvement! 👍

**Comment by MRayermannMSFT** [comment]
> I enabled it and this preserves the tabs on restart which is already a major improvement! 👍

With this setting do you still desire the rest of your feature request?

**Comment by mattmazzola** [comment]
Yes, but it is not as significant. 

It still has value if you have different sets of tabs you want open for different projects you are working on.

I think should the priority:
1. Making awareness of this "preserve explorer" setting more visible or making enabled by default
2. Allow control the text used for tab name to decrease ambiguity for multiple tabs of same container
3. (This feature) Allow saving sets of tabs as views with names
